From 6082b3764b848981c1aecdd3ae53b1b05bc25396 Mon Sep 17 00:00:00 2001
From: Guillaume Thouvenin <guillaume.thouvenin@vates.tech>
Date: Tue, 9 Jun 2026 09:36:35 +0200
Subject: tools/ocaml: silence ocaml_deprecated_auto_include alert

Ocaml's lib directory layout changed in 5.0: the unix and dynlink
libraries have been moved out of the standard library directory into
subdirectories. The compiler still locates them automatically but emits
an ocaml_deprecated_auto_include alert when doing so.

This patch sets the paths explicitly with -I +unix and -I +dynlink to
silence the alert.

Signed-off-by: Guillaume Thouvenin <guillaume.thouvenin@vates.tech>
Reviewed-by: Andrii Sultanov <andriy.sultanov@vates.tech>
Tested-by: Andrii Sultanov <andriy.sultanov@vates.tech>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>
(cherry picked from commit c3cf7c41214ac180902ca7dc826047774d3c5fb1)

diff --git a/tools/ocaml/common.make b/tools/ocaml/common.make
index c7eefceeb42b..0e6714e25a7e 100644
--- a/tools/ocaml/common.make
+++ b/tools/ocaml/common.make
@@ -11,6 +11,7 @@ OCAMLFIND ?= ocamlfind
 
 CFLAGS += -fPIC -I$(shell ocamlc -where)
 
+OCAMLINCLUDE += -I +unix -I +dynlink
 OCAMLOPTFLAGS = -g -ccopt "$(LDFLAGS)" -dtypes $(OCAMLINCLUDE) -w F -warn-error F
 OCAMLCFLAGS += -g $(OCAMLINCLUDE) -w F -warn-error F
 
